code.cpp

```diff
@@ -1,116 +1,120 @@
 #include <array>
 #include <iostream>
 #include <list>
+#include <random>
 #include <unistd.h>
+#include <vector>
 
-class Table{
+class GameOfLife {
+    private:
+        int width;
+        int height;
+        int table_size;
+        std::vector<int> table = {};
+        std::list<std::vector<int>> list_of_table;
     public:
-        static const int width = 32;
-        static const int height = 11;
-        static const int table_size =  width * height;
-};
-
-
-std::array<int,Table::table_size> step(std::array<int,Table::table_size>& arr) {
-    std::array<int, Table::table_size> res; 
-    for (int i = 0; i < Table::height; i++) {
-        for (int j = 0; j < Table::width; j++) {
-            int count = 0;
-            if (i > 0 && j > 0 && arr[Table::width * (i - 1) + (j - 1)] == 1) {
-                count++;
-            }
-            if (i > 0 && arr[Table::width * (i - 1) + j] == 1) {
-                count++;
-            }
-            if (i > 0 && j + 1 < Table::width && arr[Table::width * (i - 1) + (j + 1)] == 1) {
-                count++;
-            }
-            if (j > 0 && arr[Table::width * i + (j - 1)] == 1) {
-                count++;
-            }
-            if (j + 1 < Table::width && arr[Table::width * i + (j + 1)] == 1) {
-                count++;
+        GameOfLife(): width(32), height(11), table_size(32 * 11) {
+            table.resize(table_size);
+        };
+        GameOfLife(int width_, int height_): width(width_), height(height_), table_size(width * height) {
```

Comment:
Use width_, height_ as class private members, constructor parameters can be without _